bird_cloud_gnn/radar_dataset.py

```diff
@@ -178,6 +179,11 @@ def _process_data(self, data):
             graph.ndata["x"] = torch.tensor(local_data.values)
             graph.edata["a"] = torch.tensor(distances.data)
             self.graphs.append(graph)
+        if origin == "":
+            origin = pd.util.hash_pandas_object(data).to_string()
+        self.origin = pd.api.types.union_categoricals(
+            [self.origin, pd.Categorical([origin] * poi_indexes.shape[0])]
+        )
```

Comment:
```suggestion
        self.origin += [origin] * poi_indexes.shape[0]
```

Reply:
```suggestion
        self.origin = pd.api.types.union_categoricals(
            [self.origin, pd.Categorical([origin]).repeat(poi_indexes.shape[0])]
        )
```
We discussed offline, ignore the first code.